### PR TITLE
feat(browser): check if sign-in redirect URI has been redirected

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Provide a new method in LogtoClient to check if the sign-in redirect URI has been redirected by OIDC provider.

## Testing

- [x] Passed unit tests

### CODE REVIEW

**Review by wangsijie — COMMENTED** [review]
the naming `hasSignInRedirected` is a bit confusing? 

what about `isInSignInRedirectUri`, any idea? @logto-io/eng

**Comment by IceHe** [comment]
`isSignInRedirected`? I'm good with them.

**Comment by simeng-li** [comment]
or `isSignInRerirectUri`

File: packages/browser/src/index.test.ts [hunk 1]
```diff
@@ -166,13 +166,16 @@ describe('LogtoClient', () => {
 
     test('tokens should be set after calling signIn and handleSignInCallback successfully', async () => {
       const logtoClient = new LogtoClient({ endpoint, clientId }, requester);
+      expect(() => logtoClient.hasSignInRedirected(redirectUri)).toMatchError(
+        new LogtoClientError('sign_in_session.not_found')
+      );
```
Comment (simeng-li) [inline packages/browser/src/index.test.ts:171]: should extract to a separate test case? As the original test descriptions said `handleSignInCallback successfully`

Comment (demonzoo) [reply packages/browser/src/index.test.ts:171]: I can do that, but that would end up with many duplications. Is it necessary?
Or maybe I can slightly update the description

Comment (demonzoo) [reply packages/browser/src/index.test.ts]: Done

**Comment by demonzoo** [comment]
> `isSignInRedirected`? I'm good with them.

After group discussion, we decide to rename it to `isSignInRedirected` 👌

**Review by simeng-li — APPROVED** [review]

**Review by IceHe — APPROVED** [review]
LGTM

**Pull request merged**